src/main/java/io/cryostat/core/net/JFRConnection.java

```diff
@@ -146,6 +156,35 @@ public synchronized int getPort() {
         }
     }
 
+    public synchronized String getJvmId() throws Exception {
+        if (!isConnected()) {
+            connect();
+        }
+        ObjectName runtimeBean = new ObjectName("java.lang:type=Runtime");
+        List<String> attrNames =
+                new ArrayList<>(
+                        Arrays.asList(
+                                "ClassPath",
+                                "Name",
+                                "InputArguments",
+                                "LibraryPath",
+                                "VmVendor",
+                                "VmVersion",
+                                "StartTime"));
+
+        ByteArrayOutputStream baos = new ByteArrayOutputStream(512);
+        DataOutputStream dos = new DataOutputStream(baos);
```

Comment:
Just for good measure, should these streams be closed? A try-with-resources block would be nice.

Reply:
Yes, thanks